### PR TITLE
[logger] Allow a custom log handler as optional parameter

### PR DESCRIPTION
**Pull request by challet**
Add an extension capacity for using the native logger as a base for custom ones.

### CODE REVIEW

**Comment by domoritz** [comment]
Can you also update the types in Vega-typings?

**Comment by challet** [comment]
@domoritz sure thing, but I'm not really sure about what I'm doing here. Is there a command that I can use to check the said types ?

File: packages/vega-util/index.d.ts [hunk 1]
```diff
@@ -160,6 +160,6 @@ export const Warn: number;
 export const Info: number;
 export const Debug: number;
 
-export function logger(_?: number, method?: string): LoggerInterface;
+export function logger(_?: number, method?: string, handler?: typeof log): LoggerInterface;
 export function log(...args: readonly any[]): void;
```
Comment (challet) [inline packages/vega-util/index.d.ts:164]: Looks like the type of the `log` function used by the logger

**Comment by domoritz** [comment]
If you add a test case, you can run yarn test in the Vega typings package.

File: packages/vega-util/index.d.ts (hunk 1 above)
Comment (domoritz) [reply packages/vega-util/index.d.ts:164]: You can add a usage in https://github.com/vega/vega/tree/master/packages/vega-typings/tests/runtime and that will test the typings.

Comment (challet) [reply packages/vega-util/index.d.ts]: Thank you for pointing this out. It's done, along with expliciting these arguments types.

File: packages/vega-util/index.d.ts [hunk 2]
```diff
@@ -160,6 +160,6 @@ export const Warn: number;
 export const Info: number;
 export const Debug: number;
 
-export function logger(_?: number, method?: string): LoggerInterface;
-export function log(...args: readonly any[]): void;
+export function logger(_?: number, method?: string, handler?: typeof log): LoggerInterface;
+export function log(method: 'error' | 'warn' | 'log', level: 'error' | 'warn'| 'info' | 'debug', input: readonly any[]): void;
```
Comment (domoritz) [inline packages/vega-util/index.d.ts:164]: Should the level be all caps like in https://github.com/vega/vega/pull/3262/files#diff-377e9062f2f35efd117c4c0b89857464b8f54af1c5a8e1e648cc0683b07ca895R24?

Comment (challet) [reply packages/vega-util/index.d.ts]: That is absolutely right, sorry for the confusion

**Comment by jheer** [comment]
Thanks!

**Pull request merged**